subsys/bluetooth/audio/pbp.c

```diff
@@ -0,0 +1,188 @@
+/*
+ * Copyright 2023 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/types.h>
+
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <stdlib.h>
+
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/conn.h>
+#include <zephyr/bluetooth/gatt.h>
+
+#include <zephyr/bluetooth/audio/audio.h>
+#include <zephyr/bluetooth/audio/pbp.h>
+#include <zephyr/logging/log.h>
+
+
+
+LOG_MODULE_REGISTER(bt_pbp, CONFIG_BT_PBP_LOG_LEVEL);
+
+#define TOTAL_BUF_COUNT_NEEDED		1
+
+/* PBA Service UUID + Public Broadcast Announcement features + Maximum Metadata_Length */
+#define TOTAL_BUF_DATA_SIZE_NEEDED	260
+
+/* PBA Service UUID + Public Broadcast Announcement features + Public_Info metadata of length 1 */
+#define MIN_PBA_SIZE			5
+
+/* No audio configuration set */
+#define PBA_FEATURES_INVALID_VALUE	0
+
+NET_BUF_POOL_FIXED_DEFINE(adv_data_pool,
+			  TOTAL_BUF_COUNT_NEEDED,
+			  TOTAL_BUF_DATA_SIZE_NEEDED,
+			  CONFIG_BT_CONN_TX_USER_DATA_SIZE, NULL);
+
+static uint8_t pbp_role;
+static uint8_t stream_config_preferences = PBA_FEATURES_INVALID_VALUE;
+static struct net_buf *pba_data_buf;
+
+static void scan_check(struct bt_data *data, uint8_t *source_config)
+{
+	struct bt_uuid_16 adv_uuid;
+
+	if (data->type != BT_DATA_SVC_DATA16) {
+		return;
+	}
+
+	if (data->data_len < BT_UUID_SIZE_16 + MIN_PBA_SIZE) {
+		return;
+	}
+
+	if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
```

Comment:
Endianess?
Instead of creating `bt_uuid` you could just compare the uint16_te values.

if (sys_get_le16(data->data) != BT_UUID_PBA) return;